lib/Dredd.js

```diff
@@ -1,35 +1,94 @@
 const async = require('async');
```

Comment:
The code of this file still seems ugly to me at many places and I blame mainly the asynchronous parts. I hope with async/await/promises it could get even more readable and unit-testable in the future.